### PR TITLE
Cargo order overhaul

### PR DESCRIPTION
**Pull request by SluggyWuggies**
- Separates Armory from Security section.
- Separates Engine Construction from Engineering section.
- Combines Mining and Cargo sections. 
- Adds Livestock section.
- Adds Customes and Toys section.

Changes:
 Security: 
  - Adds Armor Crate
  - Adds Heavy Armor Crate
  - Adds Helmets Crate
  - Adds Bulletproof Helmets Crate
  - Adds Lasers Crate - Or replaces already existing one?
  - Adds Stun Batons Crate
 Armory: 
  - Adds Shotgun Ammo Crate (Why was this not already a thing?)
  - Adds R550c PDW Rifle Ammo Crate
  - Adds E-Gun Crate 
  - Riot Suppression Crate (Bundle of Armor and Shields with Batons.)
  - Adds Riot Suit Crate 
  - Adds Riot Helmet Crate
  - Adds Riot Shields Crate
  - Adds SWAT Gear Buncle Crate (Bundle of two MKISWAT Suits, two Nanotrasen SWAT Helmets, two Security Gas Masks, and two Combat Gloves.)
 Medical: 
  - Medical Bundle Crate (Box of Beakers, four Medical Spray Bottles, Box of Syringes, Box of Body Bags, two First Aid Kits, two Burn Treatment Kits, four Toxin Pills, and a Radiation Treatment Kit. What a deal!)
  - Medical Misc. Crate (Gloves, Beakers, Body Bags, Analyzers and Glasses.)
  - Surgical Supplies Crate 
 Science: 
  - Robotics Assembly Crate (Firebot, Cleanbot, and Floorbot Creation Kit.)
 Engineering:
  - Tool Crate
  - Tool Box Crate
  - Engineering Gear Crate
  - Insulated Gloves Crate
  - Low Voltage Cable Crate
  - Medium Voltage Cable Crate
  - High Voltage Cable Crate
  - Conveyor Assembly Crate
 Engine Construction: 
  - Field Generator Crate (Non-Functional. Seriously, where the hell did they go.)
  - Anti-Breach Field Generator Crate
  - Radiation Collector Crate
  - Grounding Rod Crate
 Service: 
  - Cleaning Supplies Crate (Now with soap!)
  - Exotic Carpet Crate
  - Party Equipment Crate (Wooooo!)
  - Premium Carpet Crate
 Food and Hydroponics:
  - Pizza Crate
 Livestock:
  - Bird Crate
  - Cat Crate
  - Corgi Crate
  - Chicken Crate
  - Cow Crate
  - Fox Crate
  - Pug Crate
  - Monkey Cube Crate (Individually Wrapped.)
 Mining/Cargo:
  - Cargo Supplies Crate
  - Shaft Miner Starter Kit
  - Cargo Tech Starter Kit
 Emergency: 
  - Biological Emergency Crate
  - Emergency Bot/Internals Crate
  - Explosives Emergency Crate
  - Firefighting Crate
  - Internals Crate
  - Radiation Protection Crate
  - Oxygen Canister Crate
 Miscellaneous:
  - Big Band Instrument Crate
  - Bureaucracy Crate
  - Calligraphy Crate
  - Wrapping Crate
  - Festive Wrapping Crate
  - Religious Supplies Crate
  - Functional Supplies Crate
 Customes and Toys
  - Foam Force Crate
  - Laser Tag Crate
  - Formalwear Crate
  - Original Costume Crate
  - Theatre Costume Supply Crate
  - Wizard Costume Crate
 
## Changelog
CL: adds a bunch of new orders and reorganize categories in cargo console.

### CODE REVIEW

**Comment by ThatDan123** [comment]
i dont like the huge list of items which is gonna have to be spawned for those carpet tiles crates, spawning hundreds of gameobjects will just be a lag fest

**Comment by ThatDan123** [comment]
You'll just need to remove them for now, until the cargo system supports setting stackable

**Comment by SluggyWuggies** [comment]
During some short testing, I didn't see any major issues with lag. They came out stacked in the crate, so no issues with atmos f*ckery welding the server to its frame.

**Comment by ThatDan123** [comment]
humm ok, didnt realise it supported it. Thats good.

**Comment by SluggyWuggies** [comment]
Yee. It really should be fine, though if it causes any issues, I can always change it. But from what I've seen, there shouldn't be a problem.

**Comment by ThatDan123** [comment]
Missing reference found in scriptable object CargoData field Element 0
Missing reference found in scriptable object CargoData field Element 1

**Comment by SluggyWuggies** [comment]
Hmm.. okay.

**Comment by SluggyWuggies** [comment]
Huh. Welllll. Apparently Field Generators are not a thing.

**Comment by ThatDan123** [comment]
probably just lost its reference, you'll need to put it back

**Comment by SluggyWuggies** [comment]
They disappeared around the map, too, interestingly enough. 
How would I add the reference back?

**Comment by corp-0** [comment]
click the empty field and select the field generator again

**Comment by SluggyWuggies** [comment]
What I am saying is that it is gone. Completely. From the map and from the project tab.

**Comment by SluggyWuggies** [comment]
Its a lil' wacky

**Comment by SluggyWuggies** [comment]
image

File: UnityProject/Assets/Resources/Prefabs/Items/Spawners/RandomHatPool.prefab [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: 17b2fb3ef6787a14ea8ae48d4b78cc60, type: 3}
+  m_Name: 
+  m_EditorClassIdentifier: 
+  syncMode: 0
+  syncInterval: 0.1
+  lootCount: 3
+  fanOut: 0
+  poolList:
+  - probability: 0
```
Comment (corp-0) [inline UnityProject/Assets/Resources/Prefabs/Items/Spawners/RandomHatPool.prefab:20]: probability 0 ?

Comment (SluggyWuggies) [reply UnityProject/Assets/Resources/Prefabs/Items/Spawners/RandomHatPool.prefab:20]: ah probably a mistake. i can change at next commit

**Comment by SluggyWuggies** [comment]
merging with dev doesn't fix it. dammit where did the field gens go

**Comment by Bod9001** [comment]
Missing reference found in scriptable object CargoData field Element 0
Missing reference found in scriptable object CargoData field Element 1
 cargo is missing some stuff wish it was more descriptive

**Comment by corp-0** [comment]
Are you conflicting with yourself?

**Comment by SluggyWuggies** [comment]
ignore that for now

**Comment by SluggyWuggies** [comment]
and.. probably

**Comment by SluggyWuggies** [comment]
whyyyyy conflicc

**Comment by corp-0** [comment]
you're a very conflictive person

**Comment by corp-0** [comment]
```
Found possible cargo order exploit in: 
Security/Armor Crate.
Buy price: 400 Sell price: 500

Found possible cargo order exploit in: 
Service/Janitorial Supplies Crate.
Buy price: 400 Sell price: 500

Found possible cargo order exploit in: 
Mining/Cargo/Cargo Supplies Crate.
Buy price: 350 Sell price: 500

Found possible cargo order exploit in: 
Mining/Cargo/Cargo Tech Starter Kit.
Buy price: 400 Sell price: 500

Found possible cargo order exploit in: 
Emergency/Biological Emergency Crate.
Buy price: 400 Sell price: 500

Found possible cargo order exploit in: 
Emergency/Explosive Emergency Crate.
Buy price: 400 Sell price: 502

Found possible cargo order exploit in: 
Emergency/Firefighting Crate.
Buy price: 400 Sell price: 540

Found possible cargo order exploit in: 
Emergency/Internals Crate.
Buy price: 400 Sell price: 500

Found possible cargo order exploit in: 
Miscellaneous/Wrapping Crate.
Buy price: 300 Sell price: 500

Found possible cargo order exploit in: 
Miscellaneous/Festive Wrapping Crate.
Buy price: 360 Sell price: 500
```
You didn't pass the stonks test

**Comment by SluggyWuggies** [comment]
Awwe. Okie, fixing right now

**Comment by corp-0** [comment]
you passed this time.

**Pull request merged**